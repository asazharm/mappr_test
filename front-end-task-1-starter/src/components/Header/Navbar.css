a{
    cursor: pointer;
}

.search{
    position: relative;
    /*!*top: 50%;*!*/
    /*left: 50%;*/
    /*transform: translate(-50%);*/
    transition: all 1s;
    width: 50px;
    height: 50px;
    background: white;
    box-sizing: border-box;
    border-radius: 25px;
    border: 4px solid white;
    padding: 5px;
}



.search input{
    position: absolute;
    top: 0;
    left: 0;
    width: 0;
    height: 42.5px;
    line-height: 30px;
    outline: 0;
    border: 0;
    display: none;
    font-size: 1em;
    border-radius: 20px;
    padding: 0 20px;
}

.search .fa{
    box-sizing: border-box;
    padding: 10px;
    width: 42.5px;
    height: 42.5px;
    position: absolute;
    top: 0;
    right: 0;
    border-radius: 50%;
    color: #07051a;
    text-align: center;
    font-size: 1.2em;
    transition: all 1s;
}


@media only screen and (min-width: 421px) {

    .search:hover,
    .search:valid{
        width: 200px;
        cursor: pointer;
    }

    .search:hover input,
    .search:valid input{
        transition: all 1s;
        width: 100%;
        display: block;
    }

    .search:hover .fa,
    .search:valid .fa{
        background: #07051a;
        color: white;
    }

    .search:valid a {
        display: block;
    }
}

@media only screen and (max-width: 420px) {

    .search{
        width: 100%;
        cursor: pointer;
    }

    .search input{
        transition: all 1s;
        width: 100%;
        display: block;
    }

    .search.fa{
        background: #07051a;
        color: white;
    }

    .search a {
        display: block;
    }
}

.grid_layout{
    position: absolute;
    border-radius: 30px;
    height: 80%;
    left: 43%;
    right: 43%;
    width: 14%;
    display: flex;
    background: white;
    justify-content: space-around;
    align-items: center;
}
.grid_layout-item{
    cursor: pointer;
    transition-duration: .5s;
    border-radius: 30px;
    display: flex;
    height: 90%;
    width: 30%;
    justify-content: center;
    align-items: center;
}
.grid_layout-icon{
    transition-duration: .5s;
    color: #282c34;
    height: 70%;
    width: 70%;
}
.grid_layout-item:hover, .grid_layout-item.active{
    transition-duration: .5s;
    background: #282c34;
}
.grid_layout-item:hover .grid_layout-icon, .grid_layout-item.active .grid_layout-icon{
    transition-duration: .5s;
    color: white;
}



.sort_type{
    transition-duration: .5s;
    position: absolute;
    border-radius: 30px;
    height: 80%;
    right: 110%;
    width: 35%;
    display: flex;
    background: white;
    justify-content: space-around;
    align-items: center;
}

.sort_type-item{
    transition-duration: .5s;
    border-radius: 50px;
    display: flex;
    height: 90%;
    width: 45%;
    justify-content: center;
    align-items: center;
}

.sort_type-icon{
    transition-duration: .5s;
    color: #282c34;
    height: 90%;
    width: 90%;
}

.sort_type-item:hover, .sort_type-item.active{
    transition-duration: .5s;
    background: #282c34;
}
.sort_type-item:hover .sort_type-icon, .sort_type-item.active .sort_type-icon{
    transition-duration: .5s;
    color: white;
}