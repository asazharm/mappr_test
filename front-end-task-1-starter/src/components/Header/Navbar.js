import React, { useState } from 'react';
import './Navbar.css';
import { AiOutlineFieldTime, AiOutlineLink } from 'react-icons/ai';
import { BsFillGrid3X3GapFill, BsFillGridFill, BsSquareFill } from 'react-icons/bs';
import { IoSearchSharp } from 'react-icons/io5';

function Navbar(props) {
  const [navbarShow, setNavbarShow] = useState(false);

  const handleGiphyViewChange = (event) => {
    if (event.target.id === 'favorites' && props.favorites.length === 0) alert('You don\'t have any favorites');
    else if (props.giphyViewType !== event.target.id) {
      props.setPrevGiphyViewType(props.giphyViewType);
      props.setGiphyViewType(event.target.id);
      props.setSearch('');
      setNavbarShow(false);
    }
  };

  const handleSearchChange = (event) => {
    props.setSearch(event.target.value);
    if (event.target.value > 0) {
      if (props.giphyViewType !== '') {
        props.setPrevGiphyViewType(props.giphyViewType);
        props.setGiphyViewType('');
      }
    } else {
      props.setGiphyViewType(props.prevGiphyViewType);
    }
  };

  const handleShowUploadModal = () => {
    props.setShowUploadModal(true);
  };

  const handleLayoutChange = (event) => {
    props.setLayout(event.currentTarget.id);
  };

  const handleSortChange = (event) => {
    props.setSearchSortType(event.currentTarget.id);
  };

  const navbarToggle = () => {
    setNavbarShow(!navbarShow);
  };

  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark fixed-top">
      <div className="container">
        <button className="navbar-toggler" type="button" onClick={navbarToggle}>
          <span className="navbar-toggler-icon" />
        </button>
        <div className={`collapse navbar-collapse justify-content-between${navbarShow ? ' show' : ''}`} id="navbarSupportedContent">
          <div className="navbar-nav">
            <a className={`nav-item nav-link${props.giphyViewType === 'trending' ? ' active' : ''}`} id="trending" onClick={handleGiphyViewChange}>Trending</a>
            <a className={`nav-item nav-link${props.giphyViewType === 'favorites' ? ' active' : ''}`} id="favorites" onClick={handleGiphyViewChange}>Favorites</a>
            <a className="nav-item nav-link" id="upload" onClick={handleShowUploadModal}>Upload</a>
          </div>
          {!props.isMobile
            ? (
              <div className="grid_layout">
                <div
                  className={`grid_layout-item${parseInt(props.layout) === 1 ? ' active' : ''}`}
                  id="1"
                  onClick={handleLayoutChange}
                >
                  <BsSquareFill className="grid_layout-icon" />
                </div>
                <div
                  className={`grid_layout-item${parseInt(props.layout) === 2 ? ' active' : ''}`}
                  id="2"
                  onClick={handleLayoutChange}
                >
                  <BsFillGridFill className="grid_layout-icon" />
                </div>
                <div
                  className={`grid_layout-item${parseInt(props.layout) === 3 ? ' active' : ''}`}
                  id="3"
                  onClick={handleLayoutChange}
                >
                  <BsFillGrid3X3GapFill className="grid_layout-icon" />
                </div>
              </div>
            ) : ''}
          <form className="form-inline search">
            { props.search !== '' && !props.isMobile
              ? (
                <div className="sort_type">
                  <div className={`sort_type-item${props.searchSortType === 'recent' ? ' active' : ''}`} id="recent" onClick={handleSortChange} title="Recent">
                    <AiOutlineFieldTime className="sort_type-icon" />
                  </div>
                  <div className={`sort_type-item${props.searchSortType === 'relevant' ? ' active' : ''}`} id="relevant" onClick={handleSortChange} title="Relevant">
                    <AiOutlineLink className="sort_type-icon" />
                  </div>
                </div>
              ) : ''}
            <input value={props.search} onChange={handleSearchChange} type="search" required />
            <IoSearchSharp className="fa fa-search" />
          </form>
        </div>
      </div>
    </nav>
  );
}

export default Navbar;
