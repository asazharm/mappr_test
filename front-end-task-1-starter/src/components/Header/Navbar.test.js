import React from 'react';
import renderer from 'react-test-renderer';
import Navbar from './Navbar';

test('Navbar test 1', () => {
  const component = renderer.create(
    <Navbar
      search=""
      setSearch={() => {}}
      giphyViewType="trending"
      setGiphyViewType={() => {}}
      setShowUploadModal={() => {}}
      prevGiphyViewType=""
      setPrevGiphyViewType={() => {}}
      layout={3}
      setLayout={() => {}}
      searchSortType="recent"
      setSearchSortType={() => {}}
      favorites={[]}
      isMobile={false}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Navbar test 2', () => {
  const component = renderer.create(
    <Navbar
      search="test"
      setSearch={() => {}}
      giphyViewType="search"
      setGiphyViewType={() => {}}
      setShowUploadModal={() => {}}
      prevGiphyViewType=""
      setPrevGiphyViewType={() => {}}
      layout={3}
      setLayout={() => {}}
      searchSortType="recent"
      setSearchSortType={() => {}}
      favorites={[]}
      isMobile={false}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Navbar test mobile 1', () => {
  const component = renderer.create(
    <Navbar
      search=""
      setSearch={() => {}}
      giphyViewType="trending"
      setGiphyViewType={() => {}}
      setShowUploadModal={() => {}}
      prevGiphyViewType=""
      setPrevGiphyViewType={() => {}}
      layout={3}
      setLayout={() => {}}
      searchSortType="recent"
      setSearchSortType={() => {}}
      favorites={[]}
      isMobile
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Navbar test mobile 2', () => {
  const component = renderer.create(
    <Navbar
      search="test"
      setSearch={() => {}}
      giphyViewType="search"
      setGiphyViewType={() => {}}
      setShowUploadModal={() => {}}
      prevGiphyViewType=""
      setPrevGiphyViewType={() => {}}
      layout={3}
      setLayout={() => {}}
      searchSortType="recent"
      setSearchSortType={() => {}}
      favorites={[]}
      isMobile
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
