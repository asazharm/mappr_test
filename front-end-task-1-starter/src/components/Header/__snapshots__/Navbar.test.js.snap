// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test 1 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark fixed-top"
>
  <div
    className="container"
  >
    <button
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse justify-content-between"
      id="navbarSupportedContent"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link active"
          id="trending"
          onClick={[Function]}
        >
          Trending
        </a>
        <a
          className="nav-item nav-link"
          id="favorites"
          onClick={[Function]}
        >
          Favorites
        </a>
        <a
          className="nav-item nav-link"
          id="upload"
          onClick={[Function]}
        >
          Upload
        </a>
      </div>
      <div
        className="grid_layout"
      >
        <div
          className="grid_layout-item"
          id="1"
          onClick={[Function]}
        >
          <svg
            className="grid_layout-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="16"
              rx="2"
              width="16"
            />
          </svg>
        </div>
        <div
          className="grid_layout-item"
          id="2"
          onClick={[Function]}
        >
          <svg
            className="grid_layout-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M1 2.5A1.5 1.5 0 012.5 1h3A1.5 1.5 0 017 2.5v3A1.5 1.5 0 015.5 7h-3A1.5 1.5 0 011 5.5v-3zm8 0A1.5 1.5 0 0110.5 1h3A1.5 1.5 0 0115 2.5v3A1.5 1.5 0 0113.5 7h-3A1.5 1.5 0 019 5.5v-3zm-8 8A1.5 1.5 0 012.5 9h3A1.5 1.5 0 017 10.5v3A1.5 1.5 0 015.5 15h-3A1.5 1.5 0 011 13.5v-3zm8 0A1.5 1.5 0 0110.5 9h3a1.5 1.5 0 011.5 1.5v3a1.5 1.5 0 01-1.5 1.5h-3A1.5 1.5 0 019 13.5v-3z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <div
          className="grid_layout-item active"
          id="3"
          onClick={[Function]}
        >
          <svg
            className="grid_layout-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H2a1 1 0 01-1-1V2zm5 0a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H7a1 1 0 01-1-1V2zm5 0a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1h-2a1 1 0 01-1-1V2zM1 7a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H2a1 1 0 01-1-1V7zm5 0a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H7a1 1 0 01-1-1V7zm5 0a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1h-2a1 1 0 01-1-1V7zM1 12a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H2a1 1 0 01-1-1v-2zm5 0a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H7a1 1 0 01-1-1v-2zm5 0a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1h-2a1 1 0 01-1-1v-2z"
            />
          </svg>
        </div>
      </div>
      <form
        className="form-inline search"
      >
        
        <input
          onChange={[Function]}
          required={true}
          type="search"
          value=""
        />
        <svg
          className="fa fa-search"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 428L339.92 303.9a160.48 160.48 0 0030.72-94.58C370.64 120.37 298.27 48 209.32 48S48 120.37 48 209.32s72.37 161.32 161.32 161.32a160.48 160.48 0 0094.58-30.72L428 464zM209.32 319.69a110.38 110.38 0 11110.37-110.37 110.5 110.5 0 01-110.37 110.37z"
          />
        </svg>
      </form>
    </div>
  </div>
</nav>
`;

exports[`Navbar test 2 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark fixed-top"
>
  <div
    className="container"
  >
    <button
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse justify-content-between"
      id="navbarSupportedContent"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          id="trending"
          onClick={[Function]}
        >
          Trending
        </a>
        <a
          className="nav-item nav-link"
          id="favorites"
          onClick={[Function]}
        >
          Favorites
        </a>
        <a
          className="nav-item nav-link"
          id="upload"
          onClick={[Function]}
        >
          Upload
        </a>
      </div>
      <div
        className="grid_layout"
      >
        <div
          className="grid_layout-item"
          id="1"
          onClick={[Function]}
        >
          <svg
            className="grid_layout-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="16"
              rx="2"
              width="16"
            />
          </svg>
        </div>
        <div
          className="grid_layout-item"
          id="2"
          onClick={[Function]}
        >
          <svg
            className="grid_layout-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M1 2.5A1.5 1.5 0 012.5 1h3A1.5 1.5 0 017 2.5v3A1.5 1.5 0 015.5 7h-3A1.5 1.5 0 011 5.5v-3zm8 0A1.5 1.5 0 0110.5 1h3A1.5 1.5 0 0115 2.5v3A1.5 1.5 0 0113.5 7h-3A1.5 1.5 0 019 5.5v-3zm-8 8A1.5 1.5 0 012.5 9h3A1.5 1.5 0 017 10.5v3A1.5 1.5 0 015.5 15h-3A1.5 1.5 0 011 13.5v-3zm8 0A1.5 1.5 0 0110.5 9h3a1.5 1.5 0 011.5 1.5v3a1.5 1.5 0 01-1.5 1.5h-3A1.5 1.5 0 019 13.5v-3z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <div
          className="grid_layout-item active"
          id="3"
          onClick={[Function]}
        >
          <svg
            className="grid_layout-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H2a1 1 0 01-1-1V2zm5 0a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H7a1 1 0 01-1-1V2zm5 0a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1h-2a1 1 0 01-1-1V2zM1 7a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H2a1 1 0 01-1-1V7zm5 0a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H7a1 1 0 01-1-1V7zm5 0a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1h-2a1 1 0 01-1-1V7zM1 12a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H2a1 1 0 01-1-1v-2zm5 0a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H7a1 1 0 01-1-1v-2zm5 0a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1h-2a1 1 0 01-1-1v-2z"
            />
          </svg>
        </div>
      </div>
      <form
        className="form-inline search"
      >
        <div
          className="sort_type"
        >
          <div
            className="sort_type-item active"
            id="recent"
            onClick={[Function]}
            title="Recent"
          >
            <svg
              className="sort_type-icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              t="1569683618210"
              version="1.1"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs />
              <path
                d="M945 412H689c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h256c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM811 548H689c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h122c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM477.3 322.5H434c-6.2 0-11.2 5-11.2 11.2v248c0 3.6 1.7 6.9 4.6 9l148.9 108.6c5 3.6 12 2.6 15.6-2.4l25.7-35.1v-0.1c3.6-5 2.5-12-2.5-15.6l-126.7-91.6V333.7c0.1-6.2-5-11.2-11.1-11.2z"
              />
              <path
                d="M804.8 673.9H747c-5.6 0-10.9 2.9-13.9 7.7-12.7 20.1-27.5 38.7-44.5 55.7-29.3 29.3-63.4 52.3-101.3 68.3-39.3 16.6-81 25-124 25-43.1 0-84.8-8.4-124-25-37.9-16-72-39-101.3-68.3s-52.3-63.4-68.3-101.3c-16.6-39.2-25-80.9-25-124 0-43.1 8.4-84.7 25-124 16-37.9 39-72 68.3-101.3 29.3-29.3 63.4-52.3 101.3-68.3 39.2-16.6 81-25 124-25 43.1 0 84.8 8.4 124 25 37.9 16 72 39 101.3 68.3 17 17 31.8 35.6 44.5 55.7 3 4.8 8.3 7.7 13.9 7.7h57.8c6.9 0 11.3-7.2 8.2-13.3-65.2-129.7-197.4-214-345-215.7-216.1-2.7-395.6 174.2-396 390.1C71.6 727.5 246.9 903 463.2 903c149.5 0 283.9-84.6 349.8-215.8 3.1-6.1-1.4-13.3-8.2-13.3z"
              />
            </svg>
          </div>
          <div
            className="sort_type-item"
            id="relevant"
            onClick={[Function]}
            title="Relevant"
          >
            <svg
              className="sort_type-icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M574 665.4a8.03 8.03 0 0 0-11.3 0L446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3l-39.8-39.8a8.03 8.03 0 0 0-11.3 0L191.4 526.5c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3L574 665.4zm258.6-474c-84.6-84.6-221.5-84.6-306 0L410.3 307.6a8.03 8.03 0 0 0 0 11.3l39.7 39.7c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204L665.3 562.6a8.03 8.03 0 0 0 0 11.3l39.8 39.8c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c84.5-84.6 84.5-221.5 0-306.1zM610.1 372.3a8.03 8.03 0 0 0-11.3 0L372.3 598.7a8.03 8.03 0 0 0 0 11.3l39.6 39.6c3.1 3.1 8.2 3.1 11.3 0l226.4-226.4c3.1-3.1 3.1-8.2 0-11.3l-39.5-39.6z"
              />
            </svg>
          </div>
        </div>
        <input
          onChange={[Function]}
          required={true}
          type="search"
          value="test"
        />
        <svg
          className="fa fa-search"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 428L339.92 303.9a160.48 160.48 0 0030.72-94.58C370.64 120.37 298.27 48 209.32 48S48 120.37 48 209.32s72.37 161.32 161.32 161.32a160.48 160.48 0 0094.58-30.72L428 464zM209.32 319.69a110.38 110.38 0 11110.37-110.37 110.5 110.5 0 01-110.37 110.37z"
          />
        </svg>
      </form>
    </div>
  </div>
</nav>
`;

exports[`Navbar test mobile 1 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark fixed-top"
>
  <div
    className="container"
  >
    <button
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse justify-content-between"
      id="navbarSupportedContent"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link active"
          id="trending"
          onClick={[Function]}
        >
          Trending
        </a>
        <a
          className="nav-item nav-link"
          id="favorites"
          onClick={[Function]}
        >
          Favorites
        </a>
        <a
          className="nav-item nav-link"
          id="upload"
          onClick={[Function]}
        >
          Upload
        </a>
      </div>
      
      <form
        className="form-inline search"
      >
        
        <input
          onChange={[Function]}
          required={true}
          type="search"
          value=""
        />
        <svg
          className="fa fa-search"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 428L339.92 303.9a160.48 160.48 0 0030.72-94.58C370.64 120.37 298.27 48 209.32 48S48 120.37 48 209.32s72.37 161.32 161.32 161.32a160.48 160.48 0 0094.58-30.72L428 464zM209.32 319.69a110.38 110.38 0 11110.37-110.37 110.5 110.5 0 01-110.37 110.37z"
          />
        </svg>
      </form>
    </div>
  </div>
</nav>
`;

exports[`Navbar test mobile 2 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark fixed-top"
>
  <div
    className="container"
  >
    <button
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse justify-content-between"
      id="navbarSupportedContent"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          id="trending"
          onClick={[Function]}
        >
          Trending
        </a>
        <a
          className="nav-item nav-link"
          id="favorites"
          onClick={[Function]}
        >
          Favorites
        </a>
        <a
          className="nav-item nav-link"
          id="upload"
          onClick={[Function]}
        >
          Upload
        </a>
      </div>
      
      <form
        className="form-inline search"
      >
        
        <input
          onChange={[Function]}
          required={true}
          type="search"
          value="test"
        />
        <svg
          className="fa fa-search"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 428L339.92 303.9a160.48 160.48 0 0030.72-94.58C370.64 120.37 298.27 48 209.32 48S48 120.37 48 209.32s72.37 161.32 161.32 161.32a160.48 160.48 0 0094.58-30.72L428 464zM209.32 319.69a110.38 110.38 0 11110.37-110.37 110.5 110.5 0 01-110.37 110.37z"
          />
        </svg>
      </form>
    </div>
  </div>
</nav>
`;
